addon/components/rdf-input-fields/input-field.js

```diff
@@ -44,9 +44,23 @@ export default class InputFieldComponent extends Component {
 
   get validationConstraints() {
     const { store, formGraph } = this.storeOptions;
-    return store
-      .match(this.args.field.uri, FORM('validations'), undefined, formGraph)
-      .map((t) => t.object);
+    let constraints = store.match(
+      this.args.field.uri,
+      FORM('validatedBy'),
+      undefined,
+      formGraph
+    );
+    if (constraints.length === 0) {
+      // fall back to old model:
+      constraints = store.match(
+        this.args.field.uri,
+        FORM('validations'),
+        undefined,
+        formGraph
+      );
+    }
+
+    return constraints.map((t) => t.object);
```

Comment:
indeed. it seems that the validationcontraints getter is only used in the maxLength getter which checks if there is a max length contrains. However, it then also needs the contraint's uri because it needs  the max value for the contraint... So if we want to refactor this, we need to change the helpers indeed. I will add a pr there soon to add a method returning a mapping of validation constraint uri -> validation type uri so you have both if you want

Reply:
The PR that will allow this simplification: https://github.com/lblod/submission-form-helpers/pull/61